lib/manager/maven/extract.ts

```diff
@@ -223,6 +223,61 @@ export function extractPackage(
   return result;
 }
 
+export function extractRegistries(rawContent: string): string[] {
+  if (!rawContent) {
+    return [];
+  }
+
+  const settings = parseSettings(rawContent);
+  if (!settings) {
+    return [];
+  }
+
+  const urls = [];
+
+  const mirrorUrls = parseUrls(settings, 'mirrors');
+  urls.push(...mirrorUrls);
+
+  settings.childNamed('profiles')?.eachChild((profile) => {
+    const repositoryUrls = parseUrls(profile, 'repositories');
+    urls.push(...repositoryUrls);
+  });
+
+  // filter out duplicates
+  return [...new Set(urls)];
+}
+
+function parseUrls(xmlNode: XmlElement, path: string): string[] {
+  const children = xmlNode.descendantWithPath(path);
+  const urls = [];
+  if (children?.children) {
+    children.eachChild((child) => {
```

Comment:
`children` is guaranteed to be an array if defined?

Reply:
`children` is defined as `(property) XmlElement.children: XmlNode[]`, so I would assume it to be safe to expect an array here.